### ISSUE TITLE
[Docs]: Please add that changing the `name` argument for `aws_elasticache_parameter_group` forces a replacement of the resource

### ISSUE THREAD

**Issue by abudri** [hashicorp/terraform-provider-aws#31019]
### Documentation Link

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elasticache_parameter_group

### Description

Hello,

I was changing the name of the parameter group for elasticache, and that forces a replacement. it would be nice for that to be noted in the resource documentation here:

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elasticache_parameter_group#name

`terraform plan` output:
```
  # aws_elasticache_parameter_group.redis must be replaced
-/+ resource "aws_elasticache_parameter_group" "redis" {
      ~ arn         = "arn:aws:elasticache:us-east-1:SNIP:parametergroup:cache-params" -> (known after apply)
      ~ id          = "cache-params" -> (known after apply)
      ~ name        = "cache-params" -> "redis-my-service-memory-policy" # forces replacement
```

We can see how other resource arguments note that aspect of a given argument, like here for another resource it says:

> Changing `snapshot_name` forces a new resource.

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elasticache_cluster#snapshot_name

### Would you like to implement a fix?

None

**Comment by abudri** [comment]
I can volunteer to add this to the doc too.

**Comment by AdamJCavanaugh** [comment]
I don't think this is a feasible change to put into place for all such points in the documentation. For example, changing the family would also force replacement, right? For many resources, changing the name or many (most?) required arguments force replacement of the resource. Hence the warning in plan and the ability to `state mv` when possible?
